src/utils/getDefaultDockerContext.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as cp from 'child_process';
+import { DockerOptions } from 'dockerode';
+import * as util from 'util';
+
+const exec = util.promisify(cp.exec);
+
+export async function getDefaultDockerContext(): Promise<DockerOptions | undefined> {
+    try {
+        const { stdout, stderr } = await exec('docker context inspect');
+
+        if (stderr) {
+            throw new Error(stderr);
+        }
+
+        if (!stdout) {
+            return undefined;
+        }
+
+        const dockerContext = JSON.parse(stdout);
+        // tslint:disable: no-unsafe-any
+        const host: string =
+            Array.isArray(dockerContext) && dockerContext.length > 0 &&
+            dockerContext[0].Endpoints &&
+            dockerContext[0].Endpoints.docker &&
+            dockerContext[0].Endpoints.docker.Host;
+        // tslint:enable: no-unsafe-any
+
+        if (host.indexOf('unix://') === 0) {
+            return {
+                socketPath: host.substring(7), // Everything after the unix:// (expecting unix:///var/run/docker.sock)
+            }
+        } else if (host.indexOf('npipe://') === 0) {

```

Comment:
Make these constants and use their length in the the `substring()` call?

Reply:
I thought about it, sounds good.